.github/workflows/twister.yaml

```diff
@@ -273,6 +273,19 @@ jobs:
             module_tests/twister.xml
             testplan.json
 
+      - if: matrix.subset == 1
```

Comment:
hmm, why do we need to do this on every twister run and every PR and why do we need to publish and store this data? If you want to do that, probably it needs to be done in some release related workflow (release.yaml maybe?)

Reply:
Yeah good point, I'll change the condition to only run on push. As for the artifact, yeah I guess we could run the command and leave it in the logs, but it's less step to find it has its own file.

As for the workflow, I could do it in release.yaml but then I'd have to change that workflow to use `zephyr-runner-linux-x64-4xlarge`.

Reply:
> [...]. As for the artifact, yeah I guess we could run the command and leave it in the logs, but it's less step to find it has its own file.

+1 for its own file, this is useful metadata that one may want to run a diff on, and we're looking at like just 2KB.